sql/src/main/java/io/crate/metadata/SearchPath.java

```diff
@@ -45,29 +51,40 @@ public static SearchPath createSearchPathFrom(String... schemas) {
         if (schemas == null || schemas.length == 0) {
             return new SearchPath();
         } else {
-            return new SearchPath(ImmutableList.copyOf(schemas));
+            return new SearchPath(List.of(schemas));
         }
     }
 
+    public static SearchPath createSearchPathFrom(StreamInput in) throws IOException {
+        final boolean pgCatalogIsSetExplicitly = in.readBoolean();
+        final List<String> searchPath = in.readList(StreamInput::readString);
+        return new SearchPath(pgCatalogIsSetExplicitly, searchPath);
+    }
+
     public static SearchPath pathWithPGCatalogAndDoc() {
         return PG_CATALOG_AND_DOC_PATH;
     }
 
     private SearchPath() {
-        pgCatalogIsSetExplicitly = false;
-        searchPath = ImmutableList.of(PgCatalogSchemaInfo.NAME, Schemas.DOC_SCHEMA_NAME);
+        this.pgCatalogIsSetExplicitly = false;
+        this.searchPath = ImmutableList.of(PgCatalogSchemaInfo.NAME, Schemas.DOC_SCHEMA_NAME);
```

Comment:
Could also use `List.of` given that you changed already another occurance of it.